pkg/imageverifier/image_verifier.go

```diff
@@ -0,0 +1,32 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package imageverifier
+
+import (
+	"context"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type ImageVerifier interface {
+	VerifyImage(ctx context.Context, name string, desc ocispec.Descriptor) Judgement
```

Comment:
Looking for feedback here: I chose not to include an error in the return here so plugin implementations have to decide for themselves whether they want to fail open or fail closed on errors. Does that sound reasonable?

Reply:
I have a weak opinion for including an `error`.  An `error` during execution of a verifier (for example with bindir, encountering a permission error attempting to run an executable) is different from the logic of a verification plugin issuing an explicit judgement, and an `error` allows that to be dealt with in the calling code.  An implementation of an `ImageVerifier` that chooses to fail-open might just not emit an error, or we could define an error type that is a fail-open error.

Reply:
Added an error in 0ef20624faa5f60e1778e2be4e42632eeb81b9f7.